### ISSUE TITLE
Introduce a "Reveal" action on password fields

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#6468]
Part of https://github.com/opencollective/opencollective/issues/6337
See https://dev.to/detoner777/show-hide-password-on-toggle-in-react-hooks-1lph on how to achieve that

Steps:
- [ ] Extract password input to a new `PasswordInput` component
- [ ] Update all references (in Sign in, User Security and Reset password) to make sure they all use the same component
- [ ] Introduce the "Reveal" feature, using an icon from https://styled-icons.dev/?s=eye

**Comment by Betree** [comment]
@ShreyaPrasad1209 Sure :+1:

**Comment by Betree** [comment]
Hi @ShreyaPrasad1209, do you still want to look at this one?

**Comment by ShreyaPrasad1209** [comment]
@Betree I did some analysis and it seems like creating a different component (namely `PasswordInput`) for Reveal action is not possible because it will update the state of existing `StyledInputs` and this [key](https://github.com/opencollective/opencollective-frontend/blob/add93236be98cc8d7456ba44544068c2b3fcf5ca/components/edit-collective/sections/UserSecurity.js#L313) is dependent on these specific files like UserSecurity.js. So this Reveal feature can be added inside these files and not as a separate component.

**Comment by Betree** [comment]
> @Betree I did some analysis and it seems like creating a different component (namely `PasswordInput`) for Reveal action is not possible because it will update the state of existing `StyledInputs` and this [key's value](https://github.com/opencollective/opencollective-frontend/blob/add93236be98cc8d7456ba44544068c2b3fcf5ca/components/edit-collective/sections/UserSecurity.js#L313) is dependent on specific files like UserSecurity.js, reset-password.js etc.

Doesn't sound like a blocker, we could still pass the `key` prop to this new component right?

**Comment by ShreyaPrasad1209** [comment]
> > @Betree I did some analysis and it seems like creating a different component (namely `PasswordInput`) for Reveal action is not possible because it will update the state of existing `StyledInputs` and this [key's value](https://github.com/opencollective/opencollective-frontend/blob/add93236be98cc8d7456ba44544068c2b3fcf5ca/components/edit-collective/sections/UserSecurity.js#L313) is dependent on specific files like UserSecurity.js, reset-password.js etc.
> 
> Doesn't sound like a blocker, we could still pass the `key` prop to this new component right?

Right, but they're also updating each other's state. I'll need to look into the implementation.

**Comment by shannondwray** [comment]
This would be amazing!!
